<?xml version="1.0"?>
<launch>
  <arg name="namespace" default="flir_boson"/>
  <arg name="frame_id" default="boson_camera" />
  <arg name="manager" default="$(arg frame_id)_nodelet_manager" />

  <!-- the linux file descriptor location for the camera -->
  <arg name="dev" default="/dev/video0"/>

  <!-- valid values are 30.0 or 60.0 for Bosons -->
  <arg name="frame_rate" default="60.0"/>

  <!-- valid values are RAW16 or YUV -->
  <arg name="video_mode" default="YUV"/>

  <!-- valid values are TRUE or FALSE -->
  <arg name="zoom_enable" default="FALSE"/>

  <!-- valid values are Boson_320 or Boson_640 -->
  <arg name="sensor_type" default="Boson_640"/>

  <!-- location of the camera calibration file -->
  <arg name="camera_info_url" default="package://flir_boson_usb/example_calibrations/Boson640.yaml"/>

  <group ns="$(arg namespace)">
    <!-- start nodelet manager -->
    <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" />

    <node pkg="nodelet" type="nodelet" name="$(arg manager)_driver" args="load flir_boson_usb/BosonCamera $(arg manager)">
      <param name="frame_id" type="str" value="$(arg frame_id)"/>
      <param name="dev" type="str" value="$(arg dev)"/>
      <param name="frame_rate" type="double" value="$(arg frame_rate)"/>
      <param name="video_mode" type="str" value="$(arg video_mode)"/>
      <param name="zoom_enable" type="bool" value="$(arg zoom_enable)"/>
      <param name="sensor_type" type="str" value="$(arg sensor_type)"/>
      <param name="camera_info_url" type="str" value="$(arg camera_info_url)"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="$(arg manager)_image_proc" args="load image_proc/rectify $(arg manager)">
      <remap from="image_mono" to="image_raw"/>
    </node>
  </group>
</launch>
